### PR TITLE
Refactored `call`, `deploy` and `invoke` command handlers

### PR DESCRIPTION
**Pull request by integraledelebesgue**
**Stack**:
- #2555
- #2548
- #2547
- #2546 ⬅
- #2545

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: crates/sncast/src/starknet_commands/account/add.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ pub struct Add {
     pub name: String,
 
     /// Address of the account
-    #[clap(short, long, requires = "private_key_input")]
```
Comment (cptartur) [inline crates/sncast/src/starknet_commands/account/add.rs:25]: This is breaking, not?

Comment (cptartur) [reply crates/sncast/src/starknet_commands/account/add.rs:25]: Same goes for similar changes below

**Review by kkawula — COMMENTED** [review]
Why do we remove all shortcuts for commands?

**Comment by integraledelebesgue** [comment]
> Why do we remove all shortcuts for commands?

Nevermind, global flags have been fixed on master so I reverted my changes elated to them here

File: crates/sncast/src/starknet_commands/verify.rs [hunk 2]
```diff
@@ -118,7 +118,7 @@ pub struct Verify {
     pub contract_address: Felt,
 
     /// Name of the contract that is being verified
-    #[clap(short, long)]
+    #[clap(long)]
```
[the remark this answers is not in the document]
Comment (cptartur) [reply crates/sncast/src/starknet_commands/verify.rs:121]: Please address that before merge

**Review by cptartur — APPROVED** [review]

File: crates/sncast/src/starknet_commands/verify.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (integraledelebesgue) [reply crates/sncast/src/starknet_commands/verify.rs]: Reverted this change :)

**Comment by integraledelebesgue** [comment]
✓ Commit merged in pull request [#2555](https://github.com/foundry-rs/starknet-foundry/pull/2555)

**Pull request closed without merging**